### PR TITLE
#134 lift entity browser callbacks

### LINKED ISSUE #134: Lift up EntityBrowser callbacks
`EntityBrowser` component currently is "uncontrolled". Idea is to lift up state so that parent components can pass down `SearchBar` + `EntityTable` callbacks.

## Expected Behaviour

`EntityBrowser` takes entities and callbacks as props.

## Current Behaviour

`EntityBrowser` takes only entities as props.

## Possible Solution

Update `EntityBrowserProps`:

```
interface EntityBrowserProps {
  entities: Entity[];
  onChange: (value: string) => void,
  onClear: () => void,
  onClick: () => void,
}
```

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #134.

Branched off #133.

## Description

Lifts up `EntityBrowser` callbacks to parent component.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

File: unified-codes/apps/web/src/components/templates/Explorer.tsx [hunk 1]
```diff
@@ -80,13 +79,32 @@ const mapDispatchToProps = (dispatch: Dispatch)  => {
     text: ALERT_TEXT.FETCH,
   };
 
-  const onData = (data: { entities: IEntity[] }) => dispatch(EntityActions.updateEntities(data));
-  const onError = () => dispatch(AlertActions.showAlert(alertError));
+  const onCompleted = (data: { entities: IEntity[] }) => dispatch(EntityActions.updateEntities(data));
```
Comment (mark-prins) [inline unified-codes/apps/web/src/components/templates/Explorer.tsx:82]: renamed to `ExplorerActions`

**Review by mark-prins — CHANGES_REQUESTED** [review]

File: unified-codes/apps/web/src/components/templates/Explorer.tsx (hunk 1 above)
Comment (mark-prins) [reply unified-codes/apps/web/src/components/templates/Explorer.tsx:82]: oh, and the method is changed too: `ExplorerActions.updateVariables`

File: unified-codes/apps/web/src/components/templates/Explorer.tsx [hunk 2]
```diff
@@ -80,13 +79,32 @@ const mapDispatchToProps = (dispatch: Dispatch)  => {
...
+  const onSearch = () => null;
+
+  return { onChange, onClear, onSearch, onCompleted, onError, onLoading };
 }
 
 export const Explorer = connect(mapStateToProps, mapDispatchToProps)(ExplorerComponent);
 
 export default Explorer;
+
+// const resetInput = React.useCallback(() => setInput(''), []);
```
Comment (mark-prins) [inline unified-codes/apps/web/src/components/templates/Explorer.tsx:97]: 🤔

**Comment by mark-prins** [comment]
another recommendation:

`EntityReducer` needs some updates as well:

```
import { Entity } from '@unified-codes/data';
import { EXPLORER_ACTIONS, IExplorerFetchSuccessAction } from '../actions/ExplorerActions';

const initialState = () => {
  const entities: Entity[] = [];
  return entities;
};

export const EntityReducer = (state = initialState(), action: IExplorerFetchSuccessAction) => {
  const { type } = action;

  switch (type) {
    case EXPLORER_ACTIONS.FETCH_SUCCESS: {
      const { data } = action;
      return data;
    }

    default:
      return state;
  }
};

export default EntityReducer;

```

**Comment by mark-prins** [comment]
The `Explorer.stories.tsx` also needs some rework

**Comment by katherine-sussol** [comment]
Hm, I was having a bit of trouble getting this PR running standalone (I think @mark-prins covered the two files in question above - Explorer.tsx and EntityReducer.ts) - did see some related changes in PR #140 though? Do some need to be brought in here maybe? 

Also did a @reduxjs/toolkit install manually (does this need to be added to `package.json`?

**Comment by mark-prins** [comment]
> did see some related changes in PR #138 though? Do some need to be brought in here maybe? Or better to review #138 and merge both?

Yes! well spotted 😃 
Catch 22 - this PR needs parts of the #138 - but the opening comment of that one is to wait for this one! which to do first? 🤔

**Comment by wlthomson** [comment]
Hmm, `@reduxjs/toolkit` shouldn't be a required import anymore! 

Will merge and address in next PR #138!

**Pull request merged**